packages/core/src/components/ui/ProductGallery/ProductGalleryPage.tsx

```diff
@@ -34,4 +50,4 @@ function GalleryPage({ page, title, productCard }: Props) {
   )
 }
 
-export default GalleryPage
+export default memo(ProductGalleryPage)
```

Comment:
Can we really memo this component? If the props are the same, but the selectedFacets have changed the component will not be re-rendered, right?

Reply:
I maintain my concern here, but I can be misunderstanding React. `memo` means that when a component's props don't change it won't be rerendered, right?

If that's the case we could be rendering incorrect products, if the facets have changed, since the facet is not a prop, but comes from a hook.

Reply:
Confirmed with @icazevedo that you're right about this. Will mark as resolved!